Automate builds for Windows and Mac and utilize github release system

https://github.com/features/actions

@Alcaro Heya, i'd be interested in submitting a PR for this.
I have started toying a little bit around, and currently thought of the following:
- PRs: build unoptimized GUI for win/lin/mac to ensure PR doesn't break 
- master: build optimized GUI and upload them as artifacts. (Could also create releases, if that's wished for)

Current issues I encountered, that I would appreciate help with:
- Macos straight up fails compiling on its default compiler because Apple's LLVM doesn't support `-fopenmp`. I can work around this by using llvm from homebrew, but it'd be nice if the shipped compliler would just work ootb.
- flips-gtk.cpp and flips-win32.cpp seems to be always included. For me this caused windows to always need gtk3 libs to compile.
- Using Msys2+mingw to compile on windows I have been getting `C:\Users\RUNNER~1\AppData\Local\Temp\cc5QXajF.o:libbps-suf.cpp:(*ABS*+0x0): multiple definition of '.weak.__cxa_pure_virtual._ZN4fileD2Ev'; C:\Users\RUNNER~1\AppData\Local\Temp\ccePAcuC.o:D:\a\Flips\Flips/flips.cpp:14: first defined here`
- If finding GTK libs fails when targeting GTK, it will just switch over to building CLI, instead of failing. (Isnt difficult to fix, i can do a seperate PR for this)
- Optimized builds only seem to exist for linux? Or is "just append -O3" the optimization flag for the other OS'es?

--- a/.github/workflows/build-optimized.yml
+++ b/.github/workflows/build-optimized.yml
@@ -2,6 +2,9 @@ name: Build optimized
 
 on:
   workflow_dispatch:
+  push:
+    branches:
+      - master
 
 jobs:
   linux:
@@ -44,3 +47,32 @@ jobs:
       with:
         name: macos-m1-gui.zip
         path: ./flips
+
+
+  create-release:
+    runs-on: ubuntu-latest
+    needs: [linux, windows, macos]
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Download binaries
+      uses: actions/download-artifact@v4
+      with:
+        path: builds/
+    - name: Find Flips version
+      id: flips_version
+      run: echo version=$(git rev-list --count master) >> $GITHUB_OUTPUT
+    - name: Create zips
+      run: |
+        zip -9 ./flips-linux.zip ./builds/linux-x64-gui.zip/flips
+        zip -9 ./flips-windows.zip ./builds/windows-x64-gui.zip/flips.exe
+        zip -9 ./flips-macos.zip ./builds/macos-m1-gui.zip/flips
+    - name: Upload GitHub Release
+      uses: svenstaro/upload-release-action@2.9.0
+      with:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+        file: flips-*.zip
+        file_glob: true
+        tag: v${{ steps.flips_version.outputs.version }}
+        body: "A release of commit ${{ github.sha }}."

--- a/.github/workflows/build-optimized.yml
+++ b/.github/workflows/build-optimized.yml
@@ -67,7 +67,7 @@ jobs:
       run: |
         zip -9 ./flips-linux.zip ./builds/linux-x64-gui.zip/flips
         zip -9 ./flips-windows.zip ./builds/windows-x64-gui.zip/flips.exe
-        zip -9 ./flips-macos.zip ./builds/macos-m1-gui.zip/flips
+        #zip -9 ./flips-macos.zip ./builds/macos-m1-gui.zip/flips
     - name: Upload GitHub Release
       uses: svenstaro/upload-release-action@2.9.0
       with:

--- a/README.md
+++ b/README.md
@@ -8,7 +8,7 @@ Features:
 - Can remembers which ROMs you've used, and use them again if it thinks it's correct (BPS only, GUI only)
 - Can launch other programs after patching the ROMs; together with the above, this allows you to double click a BPS to launch an emulator (GUI only)
 
-Floating IPS is in maintenance mode. Bug reports and pull requests will be read, but feature requests are unlikely to be implemented. There will be no formal releases within any forseeable future; every commit should be considered a release. If you need a version number, use the commit count.
+Floating IPS is in maintenance mode. Bug reports and pull requests will be read, but feature requests are unlikely to be implemented. Releases are automated for every commit, meaning the release schedule is highly erratic.
 
 Third-party forks, or separate tools, covering usecases this version doesn't (this only acknowledges their existence, and is not an endorsement; I haven't used most of them):
 - [Floating IPS](https://github.com/Alcaro/Flips); the original Floating IPS, in case you're currently looking at a fork

